airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams.py

```diff
@@ -0,0 +1,205 @@
+        """
+        Pydantic model to represent json schema
+        """
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def request_headers(self, *args, **kvargs) -> MutableMapping[str, Any]:
+        return {"Amazon-Advertising-API-ClientId": self._config.client_id}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an object representing single record in the response
+        """
+        yield from loads(response.text)
+
+    def get_json_schema(self):
+        return self.model.schema()
+
+
+class Profiles(AmazonAdsStream):
+    """
+    This stream corresponds to Amazon Advertising API - Profiles
+    https://advertising.amazon.com/API/docs/en-us/reference/2/profiles#/Profiles
+    """
+
+    primary_key = "profileId"
+    model = Profile
+
+    def path(self, **kvargs) -> str:
+        return "v2/profiles"
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        for record in super().parse_response(response, **kwargs):
+            profile_id_obj = self.model.parse_obj(record)
+            self.ctx.profiles.append(profile_id_obj)
+            yield record
+
+    def read_records(self, *args, **kvargs) -> Iterable[Mapping[str, Any]]:
+        if self.ctx.profiles:
+            yield from [profile.dict(exclude_unset=True) for profile in self.ctx.profiles]
+        else:
+            yield from super().read_records(*args, **kvargs)
+
+    def fill_context(self):
+        """
+        Fill profiles info for other streams in case of "profiles" stream havent been specified on catalog config
+        """
+        _ = [record for record in self.read_records(SyncMode.full_refresh)]
+
+
+class ContextStream(AmazonAdsStream):
+    """
+    Stream for getting resources based on context set by previous stream.
+    """
+
+    flattern_properties = []
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        for record in super().parse_response(response, **kwargs):
+            for prop in self.flattern_properties:
+                if prop in record:
+                    record[prop] = dumps(record[prop])
+            yield record
+
+    def read_records(self, *args, **kvargs) -> Iterable[Mapping[str, Any]]:
+        for profile in self.ctx.profiles:
+            if profile.accountInfo.type == Types.VENDOR:
+                self.ctx.current_profile_id = profile.profileId
+                yield from super().read_records(*args, **kvargs)
+
+    def request_headers(self, *args, **kvargs) -> MutableMapping[str, Any]:
+        headers = super().request_headers(*args, **kvargs)
+        headers["Amazon-Advertising-API-Scope"] = str(self.ctx.current_profile_id)
+        return headers
+
+
+class PaginationStream(ContextStream):
+    """
+    Stream for getting resources with pagination support
+    """
+
+    PAGE_SIZE = 100
```

Comment:
since this is not a constant (can be overridden in child classes)
```suggestion
    page_size = 100
```

Reply:
Fixed